objects/LuaScript.ttslua

```diff
@@ -1628,8 +1634,249 @@ function setupMiscThings()
+    label = 'Take',
+    click_function = 'takeFearButton',
+    position = { 0, 0, -2.9 },
+    width = 800,
+    height = 380,
+    font_size = 200
+  }
+  buttonZone.createButton(params)
+
+  -- take resources buttons
+  params.label = 'Take'
+  params.scale = { 2 / 1.5, 2, 2 / 1.5 }
+  params.position = { 0, 0, 1.8 }
+  for resource, guid in pairs(bags.resources) do
+    params.click_function = 'takeResource' .. resource
+    getObjectFromGUID(guid).createButton(params)
+
+    _G['takeResource' .. resource] = function(obj, clickColor, alt)
+      if not doneLoading or not isSeated(clickColor) then return end
+      takeResourceButton(resource, clickColor)
+    end
+  end
+
+  -- player resource bag counters creation
+  params = {
+    function_owner = self,
+    position = { 0, 0, 1.9 },
+    width = 700,
+    height = 700,
+    font_size = 400
+  }
+  for _, playerColor in ipairs(seatedPlayerColors) do
+    if not (options.solo and playerColor == 'Red') then
+      local vars = playerVars[playerColor]
+      for resource, guid in pairs(vars.resourceBagZones) do
+        local bag = getObjectFromGUID(vars.resourceBags[resource])
+        params.label = tostring(bag.getQuantity())
+        params.click_function = 'clickResourceBag' .. resource .. playerColor
+        getObjectFromGUID(guid).createButton(params)
+
+        _G['clickResourceBag' .. resource .. playerColor] = function(obj, clickColor, alt)
+          if not doneLoading or clickColor ~= playerColor then return end
+          if alt then
+            if bag.getQuantity() > 0 then getObjectFromGUID(vars.trashGuid).putObject(bag.takeObject()) end
+          else
+            takeResourceButton(resource, clickColor)
+          end
+        end
+      end
+      if options.playerBoards[playerColor] == 4 then
+        for resource, guid in pairs(suitcaseResourceBags) do
+          local bag = getObjectFromGUID(guid)
+          if bag ~= nil then
+            params.label = tostring(bag.getQuantity())
+            params.click_function = 'clickResourceBag' .. resource .. 'Suitcase'
+            getObjectFromGUID(vars.suitcaseResourceBagZones[resource]).createButton(params)
+
+            _G['clickResourceBag' .. resource .. 'Suitcase'] = function(obj, clickColor, alt)
+              if not doneLoading or clickColor ~= playerColor then return end
+              if alt then
+                if bag.getQuantity() > 0 then getObjectFromGUID(vars.trashGuid).putObject(bag.takeObject()) end
+              else
+                takeResourceButton(resource, clickColor, true)
+              end
+            end
+          end
+        end
+      end
+    end
+  end
+
+  -- shuffle bonus encounter cards and reposition them to make space for the chapter 6 deck
+  for i = 1, 5 do
+    local zone = getObjectFromGUID(deckZones["bonus" .. i])
+    for _, obj in ipairs(zone.getObjects()) do
+      if obj.type == "Deck" then
+        obj.setPosition(obj.getPosition() + vector(-4.0, 0, 0))
```

Comment:
I'd rather have the decks at the same position always and just place the 6th deck next to them

Reply:
The movement of the decks is to better account the space for the 6th deck. I would prefer to move the existing 5 a bit to the left as it looks kind of unbalanced otherwise when you have 6 decks. I don't think it becomes an issue as nothing in campaign mode will use the script zones or anything there after this movement.

Reply:
What if the 6th deck was below the 3rd one for symmetry?

Reply:
I would still prefer to keep this 6th deck where it is currently. The space below the decks are relatively busy and logically it make sense and the space is there for the 6th deck to go where it is currently.

Reply:
I just think it's really weird to have zones for the decks and then move them away from them 😅

Reply:
I get that haha but the zones are being used for their intended purposes imo. In non campaign, they’re being used throughout the script and thus the cards stay where they are. In campaign mode they’re only used as part of the initial script so they’re useless after that.

Reply:
Okay, I'll give in :-P